pkg/docker/docker.go

```diff
@@ -117,36 +115,37 @@ type Docker interface {
 	DownloadFromContainer(containerPath string, w io.Writer, container string) error
-	Ping() error
+	Version() (dockertypes.Version, error)
 }
 
-// Client contains all methods used when interacting directly with docker engine-api instead of the k8s abstraction around docker engine-api
+// Client contains all methods used when interacting directly with docker engine-api
 type Client interface {
 	ContainerAttach(ctx context.Context, container string, options dockertypes.ContainerAttachOptions) (dockertypes.HijackedResponse, error)
-	ContainerWait(ctx context.Context, containerID string) (int, error)
 	ContainerCommit(ctx context.Context, container string, options dockertypes.ContainerCommitOptions) (dockertypes.ContainerCommitResponse, error)
+	ContainerCreate(ctx context.Context, config *dockercontainer.Config, hostConfig *dockercontainer.HostConfig, networkingConfig *dockernetwork.NetworkingConfig, containerName string) (dockertypes.ContainerCreateResponse, error)
+	ContainerInspect(ctx context.Context, containerID string) (dockertypes.ContainerJSON, error)
+	ContainerRemove(ctx context.Context, containerID string, options dockertypes.ContainerRemoveOptions) error
+	ContainerStart(ctx context.Context, containerID string) error
+	ContainerWait(ctx context.Context, containerID string) (int, error)
 	CopyToContainer(ctx context.Context, container, path string, content io.Reader, opts dockertypes.CopyToContainerOptions) error
 	CopyFromContainer(ctx context.Context, container, srcPath string) (io.ReadCloser, dockertypes.ContainerPathStat, error)
 	ImageBuild(ctx context.Context, buildContext io.Reader, options dockertypes.ImageBuildOptions) (dockertypes.ImageBuildResponse, error)
 	ImageInspectWithRaw(ctx context.Context, imageID string, getSize bool) (dockertypes.ImageInspect, []byte, error)
+	ImagePull(ctx context.Context, ref string, options dockertypes.ImagePullOptions) (io.ReadCloser, error)
+	ImageRemove(ctx context.Context, imageID string, options dockertypes.ImageRemoveOptions) ([]dockertypes.ImageDelete, error)
+	ServerVersion(ctx context.Context) (dockertypes.Version, error)
 }
 
 type stiDocker struct {
-	kubeDockerClient dockertools.DockerInterface
-	client           Client
-	httpClient       *http.Client
-	dialer           *net.Dialer
-	pullAuth         dockertypes.AuthConfig
-	endpoint         string
+	client   Client
+	pullAuth dockertypes.AuthConfig
 }
 
-func (s stiDocker) InspectImage(name string) (*dockertypes.ImageInspect, error) {
-	ctx, _ := context.WithTimeout(context.Background(), 2*time.Minute)
-	resp, _, err := s.client.ImageInspectWithRaw(ctx, name, true)
+func (d stiDocker) InspectImage(name string) (*dockertypes.ImageInspect, error) {
+	ctx, cancel := getDefaultContext()
+	defer cancel()
+	resp, _, err := d.client.ImageInspectWithRaw(ctx, name, true)
 	if err != nil {
-		if dockerapi.IsErrImageNotFound(err) {
-			return nil, fmt.Errorf("no such image :%q", name)
-		}
```

Comment:
having this method return an error that contained the string "no such image" was critical to getting a bunch of test flows working in the first place, is the d.client.ImageInspectWithRaw() now handling that for us?  Callers to inspectImage need a way to distinguish between a legitimate error and an image not found error.